xray/default_streaming_strategy.go

```diff
@@ -64,12 +66,15 @@ func (dSS *DefaultStreamingStrategy) StreamCompletedSubsegments(seg *Segment) []
 		seg.Subsegments[len(seg.Subsegments)-1] = nil
 		seg.Subsegments = seg.Subsegments[:len(seg.Subsegments)-1]
 
-		seg.ParentSegment.totalSubSegments--
+		atomic.AddUint32(&seg.ParentSegment.totalSubSegments, ^uint32(0))
```

Comment:
What does the `^` mean in front of the `uinit32`?

Reply:
This means to decrement the value by 1 : [reference](https://golang.org/src/sync/atomic/doc.go?s=3850:3905#L88)